File: native/android/PropertyCross/src/com/propertycross/android/presenter/PropertyPresenter.java
package com.propertycross.android.presenter;

import com.propertycross.android.events.Callback;
import com.propertycross.android.events.UIEvent;
import com.propertycross.android.model.Property;

public class PropertyPresenter {

	public interface View {
		void setProperty(Property property);
		void setIsFavourited(boolean isFavourited);
		void toggleFavourite(Callback<UIEvent> callback);
	}

	private Property property;
	private PropertyFinderPersistentState state;
	private View view;

	public PropertyPresenter(PropertyFinderPersistentState state, Property property) {
		this.state = state;
		this.property = property;
	}

	public void setView(View v) {
		view = v;
		view.setProperty(property);
		view.toggleFavourite(new Callback<UIEvent>() {
			
			public void complete(UIEvent event) {
				state.toggleFavourite(property);
				view.setIsFavourited(state.isPropertyFavourited(property));
			}
		});
		view.setIsFavourited(state.isPropertyFavourited(property));
	}
}

File: xamarin/common/Presenter/PropertyPresenter.cs
using PropertyCross.Model;
using System;

namespace PropertyCross.Presenter
{
  /// <summary>
  /// A presenter that renders a single property
  /// </summary>
  public class PropertyPresenter
  {
    /// <summary>
    /// The interface this presenter requires from the assocaited view.
    /// </summary>
    public interface View
    {
      void SetProperty(Property property);

      bool IsFavourited { set; }

      event EventHandler ToggleFavourite;
    }

    private Property _property;

    private PropertyCrossPersistentState _state;

    private View _view;

    public PropertyPresenter(PropertyCrossPersistentState state, Property property)
    {
      _state = state;
      _property = property;
    }

    public void SetView(View view)
    {
      _view = view;

      view.SetProperty(_property);
      view.ToggleFavourite += View_ToggleFavourite;
      view.IsFavourited = _state.IsPropertyFavourited(_property);
    }

    private void View_ToggleFavourite(object sender, EventArgs e)
    {
      _state.ToggleFavourite(_property);
      _view.IsFavourited = _state.IsPropertyFavourited(_property);
    }
  }
}
